src/library/scala/collection/immutable/HashMap.scala

```diff
@@ -463,11 +495,11 @@ private final class BitmapIndexedMapNode[K, +V](
       val key0UnimprovedHash = getHash(index)
       if (key0UnimprovedHash == originalHash && key0 == key) {
         val value0 = this.getValue(index)
-        return (
+        return {
```

Comment:
Normally I'm a fan of early return, but the returns here aren't actually early!  There's nothing that can't be accomplished without adding a couple extra `else`s.  While you're in here, do you mind rewriting this so it's more comprehensible?

Reply:
Fixed